public/js/toggleCompletedCourses.js

```diff
@@ -1,3 +1,19 @@
 document.querySelector('.completed-courses-btn').addEventListener('click', () => {
     document.querySelector('#completed-courses').classList.toggle('hidden')
-})
\ No newline at end of file
+})
+
+
+document.addEventListener('DOMContentLoaded', function() {
+    const dropdown = document.querySelector('.courseCategoryFilter');
+    // Set the dropdown based on the URL parameter
+    const urlParams = new URLSearchParams(window.location.search);
+    const categoryParam = urlParams.get('category');
+    if (categoryParam) {
+        dropdown.value = categoryParam;
+    }
+    // Event listener for changing the dropdown
+    dropdown.addEventListener('change', () => {
+        const selectedValue = dropdown.value;
+        window.location.href = `http://0.0.0.0:8080/courses?category=${selectedValue}`;
```

Comment:
When this project is deployed onto a server, the address won't be `0.0.0.0:8080` any more, so you need to use a relative address here.